test/io.jl

```diff
@@ -41,13 +41,18 @@ module TestIO
                    B = 'a':'c',
                    C = ["A", "B", "C"],
                    D = CategoricalArray('a':'c'),
-                   E = NullableCategoricalArray(["A", "B", "C"]),
-                   E = Vector{?Int}(1:3),
-                   F = nulls(3),
-                   G = fill(null, 3))
+                   E = CategoricalArray(["A", "B", null]),
+                   F = Vector{?Int}(1:3),
+                   G = nulls(3),
+                   H = fill(null, 3))
 
-    answer = Sys.WORD_SIZE == 64 ? 0x3bd4a4a099ae8aac : 0x5ecdc4bb
-    @test hash(sprint(printtable, dt)) == answer
+    DRT = CategoricalArrays.DefaultRefType
+    @test sprint(printtable, dt) == """
+                                    "A","B","C","D","E","F","G","H"
+                                    1,"a","A","CategoricalArrays.CategoricalValue{Char,$DRT} 'a'","CategoricalArrays.CategoricalValue{String,$DRT} "A"","1",NULL,NULL
```

Comment:
Actually, there's been a change in 0.7 recently. Array printing used to set `:compact => true`, and that's no longer the case. The 0.6 behavior is better: it only prints the string when printing a categorical array, but prints the full details when showing a single value. Better tackle this as a separate issue. For now, probably better base print tests on the 0.6 output, which is more or less what we should be aiming for in 0.7 too.